src/python/tests/core/api/test_browser_assets.py

```diff
@@ -0,0 +1,635 @@
+        Verify that if more than 20 readings, summary of only last n hrs readings for a sensor value are returned when
+        hours is passed as query parameter
+        http://localhost:8082/foglamp/asset/TESTAPI/x?hours=2
+        """
+        conn = http.client.HTTPConnection(BASE_URL)
+        conn.request("GET", '/foglamp/asset/{}/{}/summary?hours={}'.format(test_data_asset_code, sensor_code_1, 2))
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        conn.close()
+        retval = json.loads(r)
+        assert 1 == len(retval)
+        assert retval[sensor_code_1]['min'] == min(self.test_data_x_val_list[-3:])
+        assert retval[sensor_code_1]['average'] == \
+               sum(self.test_data_x_val_list[-3:]) / len(self.test_data_x_val_list[-3:])
+        assert retval[sensor_code_1]['max'] == max(self.test_data_x_val_list[-3:])
+
+    @pytest.mark.xfail(reason="FOGL-546")
+    async def test_get_asset_sensor_readings_stats_q_time_complex(self):
+        """
+        Verify that if a combination of hrs, min, sec is used, shortest period will apply for sensor reading
+        http://localhost:8082/foglamp/asset/TESTAPI/x/summary?hours=20&minutes=20&seconds=180&limit=20
+        """
+        conn = http.client.HTTPConnection(BASE_URL)
+        conn.request("GET", '/foglamp/asset/{}/{}/summary?hours={}&minutes={}&seconds={}&limit={}'
+                     .format(test_data_asset_code, sensor_code_1, 20, 20, 180, 20))
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        conn.close()
+        retval = json.loads(r)
+        assert 1 == len(retval)
+        assert retval[sensor_code_1]['min'] == self.test_data_x_val_list[-1]
+        assert retval[sensor_code_1]['average'] == self.test_data_x_val_list[-1]
+        assert retval[sensor_code_1]['max'] == self.test_data_x_val_list[-1]
+
+    @pytest.mark.xfail(reason="FOGL-546")
+    async def test_get_asset_sensor_readings_time_avg(self):
+        """
+        Verify that series data is grouped by default on seconds
+        http://localhost:8082/foglamp/asset/TESTAPI/x/series
+        """
+        conn = http.client.HTTPConnection(BASE_URL)
+        conn.request("GET", '/foglamp/asset/{}/{}/series'.format(test_data_asset_code, sensor_code_1))
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        conn.close()
+        retval = json.loads(r)
+        # Find unique set of times grouped by seconds from test data
+        grouped_ts_sec = self.group_date_time(unit="second")
+
+        # Verify the length of groups and value of last element
+        assert len(grouped_ts_sec) == len(retval)
+        assert retval[-1]["average"] == self.test_data_x_val_list[-1]
+        assert retval[-1]["max"] == self.test_data_x_val_list[-1]
+        assert retval[-1]["min"] == self.test_data_x_val_list[-1]
+        assert retval[-1]["time"] == grouped_ts_sec[-1]
```

Comment:
avg, min and max of 1 value will be that itself! what I'm missing here?

Reply:
retval returns n values grouped by seconds. We are verifying only the last result and since we have only 1 record in that second, we are verifying that.